### PR TITLE
Postgres: Convert tests to stdlib

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Convert Postgres data source tests to standard library + testify, to ease reviewing of @ying-jeanne's PR.

I would recommend diffing using a graphical tool, since it's much easier to see actual changes.

### CODE REVIEW

File: pkg/tsdb/postgres/postgres_test.go [hunk 1]
```diff
@@ -216,14 +216,14 @@ func TestPostgres(t *testing.T) {
...
-			require.Equal(t, 1.1, column[5].(float64))
-			require.Equal(t, 1.2, column[6].(float64))
+			require.Equal(t, int64(1), column[0].(int64))
+			require.Equal(t, int64(2), column[1].(int64))
+			require.Equal(t, int64(3), column[2].(int64))
+
+			require.Equal(t, int64(4.5), column[3].(float64))
+			require.Equal(t, int64(6.7), column[4].(float64))
+			require.Equal(t, int64(1.1), column[5].(float64))
+			require.Equal(t, int64(1.2), column[6].(float64))
```
Comment (sakjur) [inline pkg/tsdb/postgres/postgres_test.go]:
```suggestion
			require.Equal(t, float64(4.5), column[3].(float64))
			require.Equal(t, float64(6.7), column[4].(float64))
			require.Equal(t, float64(1.1), column[5].(float64))
			require.Equal(t, float64(1.2), column[6].(float64))
```

Comment (aknuds1) [reply pkg/tsdb/postgres/postgres_test.go]: @sakjur Fixed. Have only one bug left.

**Review by marefr — COMMENTED** [review]
Looks great :+1: Just fix the failing integration tests and we can merge this.

**Comment by aknuds1** [comment]
@marefr Thanks! Fixed the last one, was because of a test bleeding state into it siblings.

**Review by kylebrandt — APPROVED** [review]
<3

**Pull request merged**